File: spring-web-java/src/main/java/spring/web/java/domain/repository/OrderRepository.java
package spring.web.java.domain.repository;

import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import spring.web.java.domain.model.entity.Order;
import spring.web.java.domain.model.enums.OrderStatus;

public interface OrderRepository {

	Page<Order> findAll(Pageable pageable, Long memberId, OrderStatus orderStatus);

	Optional<Order> findById(Long id);

	Order save(Order order);
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/repository/OrderRepository.kt
package spring.web.kotlin.domain.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import spring.web.kotlin.domain.model.entity.Order
import spring.web.kotlin.domain.model.enums.DeliveryStatus
import spring.web.kotlin.domain.model.enums.OrderStatus

interface OrderRepository {
    fun findAll(
        pageable: Pageable,
        memberId: Long?,
        orderStatus: OrderStatus?,
        deliveryStatus: DeliveryStatus?
    ): Page<Order>

    fun findByIdOrNull(id: Long): Order?
    fun save(order: Order): Order
}